Wrappers/Python/cil/framework/framework.py

```diff
@@ -2467,306 +2504,198 @@ def __ne__(self, other):
         if isinstance(other, DataContainer):
             return self.as_array()!=other.as_array()
         return self.as_array()!=other      
-    
-    @property
-    def size(self):
-        '''Returns the number of elements of the DataContainer'''
-        return self.as_array().size
-
-    @property
-    def dtype(self):
-        '''Returns the type of the data array'''
-        return self.as_array().dtype
-    
+        
 class ImageData(DataContainer):
     '''DataContainer for holding 2D or 3D DataContainer'''
     __container_priority__ = 1
-    
-    
+
+    @property
+    def geometry(self):
+        return self.__geometry
+
+    @geometry.setter
+    def geometry(self, val):
+        self.__geometry = val
+
+    @property
+    def dimension_labels(self):
+        return self.geometry.dimension_labels
+      
+    @dimension_labels.setter
+    def dimension_labels(self, val):
+        if val is not None:
+            print("use geometry.set_labels() to set the dimension_labels")
```

Comment:
I think we should crash rather than print a message, as a user can do this operation and not realise that the system didn't do what they wanted.